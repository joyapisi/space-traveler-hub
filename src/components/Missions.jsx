/* eslint-disable camelcase */
import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getMissions, joinMission, cancelMission } from '../redux/missions/missionSlice';

const Missions = () => {
  const { missions } = useSelector((store) => store.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    if (missions.length === 0) dispatch(getMissions());
  }, [dispatch, missions.length]);

  const handleJoinMission = (mission_id) => {
    dispatch(joinMission(mission_id));
  };

  const handleCancelMission = (mission_id) => {
    dispatch(cancelMission(mission_id));
  };

  return (
    <div className="missions-container">
      <table>
        <thead>
          <tr className="tableHeader-row">
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.mission_id} className="mission-row">
              <td className="mission-name">{mission.mission_name}</td>
              <td className="mission-description">{mission.description}</td>
              <td className="mission-reservation">
                {mission.reserved ? (
                  <button className="member-button" type="button">
                    Active Member
                  </button>
                ) : (
                  <button className="member-button" type="button">
                    NOT A MEMBER
                  </button>
                )}
              </td>
              <td className="mission-reservation">
                {mission.reserved ? (
                  <button
                    className="mission-button"
                    type="button"
                    onClick={() => handleCancelMission(mission.mission_id)}
                  >
                    Leave Mission
                  </button>
                ) : (
                  <button
                    className="mission-button"
                    type="button"
                    onClick={() => handleJoinMission(mission.mission_id)}
                  >
                    Join Mission
                  </button>
                )}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
