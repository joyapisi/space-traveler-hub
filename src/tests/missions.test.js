import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import Missions from '../components/Missions';
import Store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Missions', () => {
  test('Render Missions Component', () => {
    const tree = TestRenderer.create(
      <Provider store={Store}>
        <Missions />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
