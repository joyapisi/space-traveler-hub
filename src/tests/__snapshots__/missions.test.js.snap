// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Render Missions Component 1`] = `
<div
  className="missions-container"
>
  <table>
    <thead>
      <tr
        className="tableHeader-row"
      >
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
